de.dlr.sc.virsat.fdir.core/src/de/dlr/sc/virsat/fdir/core/matrix/IMatrix.java

```diff
@@ -33,4 +32,44 @@
 	 * @return MatrixIterator
 	 */
 	MatrixIterator getIterator(double[] probabilityDistribution, double delta, double eps);
+
+	/**
+	 * @return returns number of states
+	 */
+	int getCountStates();
+
+	/**
+	 * @return returns matrix diagonal
+	 */
+	double[] getDiagonal();
+
+	/**
+	 * @return returns StatePredIndices
+	 */
+	int[][] getStatePredIndices();
+
+	/**
+	 * @return return StatePredRates
+	 */
+	double[][] getStatePredRates();
+
+	/**
+	 * @param diagonal sets new diagonal for this matrix
+	 */
+	void setDiagonal(double[] diagonal);
+
+	/**
+	 * @param statePredIndices sets new StatePredIndices
+	 */
+	void setStatePredIndices(int[][] statePredIndices);

```

Comment:
I think we can remove this from the interface.

Reply:
I think we still need this in the interface. We could come up with a better, more generic name though.